src/pytest_mock_resources/fixture/database/redis.py

```diff
@@ -31,15 +44,25 @@ def create_redis_fixture(scope="function"):
     """
 
     @pytest.fixture(scope=scope)
-    def _(_redis_container, pmr_redis_config):
-        db = redis.Redis(host=pmr_redis_config.host, port=pmr_redis_config.port)
-        db.flushall()
+    def _(request, _redis_container, pmr_redis_config):
+        database_number = 0
+        if hasattr(request.config, 'workerinput'):
+            worker_input = request.config.workerinput
+            worker_id = worker_input['workerid']  # For example "gw0".
+            database_number = int(worker_id[2:])
```

Comment:
Another way to do this would be to   store databases created in database 0-

Init database 0 with a key value pair for every database created so far, starting with 1

Increment until 15

Not necessarily better but similar to the way it works for Postgres. That way there is also no dependencies on worker impl and support for parallelization is generic.

Reply:
we could do this, if we were willing to lose a database to bookkeeping. I think the tradeoff isn't great. Also the way you get that in redis with transactions/atomicity is going to be a lot more tricky also.

tbh, i see this as a stopgap for a future system where we track the data in a filelock per my other pr. and/or spin up N redis containers on different ports instead of sharing one

Reply:
👍